app_flutter/lib/task_box.dart

```diff
@@ -215,38 +212,119 @@ class TaskOverlayContents extends StatelessWidget {
-            ),
+          child: TaskOverlayContents(
+            showSnackbarCallback: Scaffold.of(parentContext).showSnackBar,
+            buildState: buildState,
+            task: task,
+            taskStatus: taskStatus,
           ),
         ),
       ],
     );
   }
 }
+
+/// Displays the information from [Task] and allows interacting with a [Task].
+///
+/// This is intended to be inserted in [TaskOverlayEntry].
+///
+/// Offers the functionality of opening the log for this [Task] and rerunning
+/// this [Task] through the build system.
+class TaskOverlayContents extends StatelessWidget {
+  const TaskOverlayContents({
+    Key key,
+    @required this.showSnackbarCallback,
+    @required this.buildState,
+    @required this.task,
+    @required this.taskStatus,
+  })  : assert(showSnackbarCallback != null),
+        assert(buildState != null),
+        assert(task != null),
+        super(key: key);
+
+  final ScaffoldFeatureController<SnackBar, SnackBarClosedReason> Function(
+      SnackBar) showSnackbarCallback;
+
+  /// A reference to the [FlutterBuildState] for performing operations on this [Task].
+  final FlutterBuildState buildState;
+
+  /// The [Task] to display in the overlay
+  final Task task;
+
+  /// [Task.status] modified to take into account [Task.attempts] to create
+  /// a more descriptive status.
+  final String taskStatus;
+
+  @visibleForTesting
+  static const String rerunErrorMessage = 'Failed to rerun task.';
+  @visibleForTesting
+  static const String rerunSuccessMessage =
+      'Devicelab is rerunning the task. This can take a minute to propagate.';
+  @visibleForTesting
+  static const Duration rerunSnackbarDuration = Duration(seconds: 15);
+
+  /// A lookup table to define the [Icon] for this [taskStatus].
+  static const Map<String, Icon> statusIcon = <String, Icon>{
+    TaskBox.statusFailed: Icon(Icons.clear, color: Colors.red, size: 32),
+    TaskBox.statusNew: Icon(Icons.new_releases, color: Colors.blue, size: 32),
+    TaskBox.statusInProgress:
+        Icon(Icons.autorenew, color: Colors.blue, size: 32),
+    TaskBox.statusSucceeded:
+        Icon(Icons.check_circle, color: Colors.green, size: 32),
+    TaskBox.statusSucceededButFlaky: Icon(Icons.check_circle_outline, size: 32),
+    TaskBox.statusUnderperformed:
+        Icon(Icons.new_releases, color: Colors.orange, size: 32),
+    TaskBox.statusUnderperformedInProgress:
+        Icon(Icons.autorenew, color: Colors.orange, size: 32),
+  };
+
+  @override
+  Widget build(BuildContext context) {
+    return Card(
+      child: Column(
+        mainAxisSize: MainAxisSize.max,
+        children: <Widget>[
+          ListTile(
+            leading:
+                Tooltip(message: taskStatus, child: statusIcon[taskStatus]),
+            title: Text(task.name),
+            subtitle: Text(
+                'Attempts: ${task.attempts}\nDuration: ${task.endTimestamp - task.startTimestamp} seconds\nAgent: ${task.reservedForAgentId}'),
```

Comment:
This is getting a bit long - you can split a string onto multiple lines like so:

```
var x = 'This is all '
  'part of one String';
```

Reply:
Thanks for sharing! This is a cool feature.

Done.